### PR TITLE
feat(api-server): extend Inspect API with new ResourceRules

### PR DESCRIPTION
**Pull request by Automaat**
### Checklist prior to review

- [x] [Link to relevant issue][1] as well as docs and UI issues --
- [x] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [x] Tests (Unit test, E2E tests, manual test on universal and k8s) --
  - Don't forget `ci/` labels to run additional/fewer tests
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label) --

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: pkg/api-server/testdata/resources/inspect/dataplanes/_rules/meshgateway.golden.json [hunk 1]
```diff
@@ -12,6 +12,40 @@
...
+       "resourceMeta": {
+        "labels": {
+         "k8s.kuma.io/namespace": "kuma-demo",
+         "kuma.io/display-name": "mt-on-gateway"
+        },
+        "mesh": "default",
+        "name": "mt-on-gateway",
+        "type": "Mesh"
+       },
+       "ruleIndex": 0
```
Comment (lobkovilya) [inline pkg/api-server/testdata/resources/inspect/dataplanes/_rules/meshgateway.golden.json:37]: could you please add a test when `ruleIndex` is greater than 0?

File: pkg/api-server/testdata/resources/inspect/dataplanes/_rules/meshgateway.golden.json [hunk 2]
```diff
@@ -12,6 +12,40 @@
...
+      }
+     ],
+     "resourceMeta": {
+      "labels": {},
+      "mesh": "",
+      "name": "default",
+      "type": "Mesh"
+     },
+     "resourceSectionName": ""
```
Comment (lobkovilya) [inline pkg/api-server/testdata/resources/inspect/dataplanes/_rules/meshgateway.golden.json:46]: and also we need a test when `resourceSectionName` is set

File: pkg/api-server/testdata/resources/inspect/dataplanes/_rules/meshgateway.golden.json (hunk 1 above)
Comment (Automaat) [reply pkg/api-server/testdata/resources/inspect/dataplanes/_rules/meshgateway.golden.json:37]: added

File: pkg/api-server/testdata/resources/inspect/dataplanes/_rules/meshgateway.golden.json (hunk 2 above)
Comment (Automaat) [reply pkg/api-server/testdata/resources/inspect/dataplanes/_rules/meshgateway.golden.json:46]: added

File: docs/generated/openapi.yaml [hunk 3]
```diff
@@ -2416,6 +2416,38 @@ components:
...
+        conf:
+          description: The actual conf generated
+          type: object
+          x-go-type: interface{}
+        origin:
+          type: array
+          description: >
+            The list of policies that contributed to the 'conf'. The order is
+            important as it reflects
+
```
[the remark this answers is not in the document]
Comment (Automaat) [reply docs/generated/openapi.yaml:2447]: I tried changing this, but `make check` is overriding this to look like this

[the remark this answers is not in the document]
Comment (lobkovilya) [reply docs/generated/openapi.yaml:2447]: you need to change to one line here https://github.com/kumahq/kuma/pull/11040/files#diff-c732b067e6952dd1ff85f2d63513bc3ebf5d75bac077965ec18d44dfff53140aR177

[the remark this answers is not in the document]
Comment (Automaat) [reply docs/generated/openapi.yaml:2447]: This is what I was changin, and `make check` was overriding it

File: pkg/api-server/testdata/resources/inspect/dataplanes/_rules/meshgateway.golden.json [hunk 4]
```diff
@@ -21,7 +55,7 @@
        "requestTimeout": "10s"
       }
      },
-     "matchers": [],
+     "matchers": null,
```
Comment (lukidzi) [inline pkg/api-server/testdata/resources/inspect/dataplanes/_rules/meshgateway.golden.json:58]: Can it break UI? We used to return empty list and now it's nil

Comment (lahabana) [inline pkg/api-server/testdata/resources/inspect/dataplanes/_rules/meshgateway.golden.json:58]: We should always have empty arrays and not nil in API result

File: pkg/api-server/testdata/resources/inspect/dataplanes/_rules/meshgateway.golden.json [hunk 5]
```diff
@@ -63,6 +97,7 @@
      }
     ]
    },
+   "toResourceRules": null,
```
Comment (lahabana) [inline pkg/api-server/testdata/resources/inspect/dataplanes/_rules/meshgateway.golden.json:100]: We should always have empty arrays and not nil in API result

**Review by lobkovilya — APPROVED** [review]

**Pull request merged**